frontend/cypress/integration/common/graph_context_menu.ts

```diff
@@ -0,0 +1,32 @@
+import { Then, When } from 'cypress-cucumber-preprocessor/steps';
+
+When('user opens the context menu of the {string} service node', function (svcName: string) {
+  cy.waitForReact();
+  cy.getReact('CytoscapeGraph')
+    .should('have.length', '1')
+    .getCurrentState()
+    .then(state => {
+      const node = state.cy.nodes(`[nodeType="service"][service="${svcName}"]`);
+      node.emit('cxttapstart');
+    });
+});
+
+When('user clicks the {string} item of the context menu', function (menuKey: string) {
+  cy.get(`[data-test="graph-node-context-menu"] [data-test="${menuKey}"]`).then($item => {
+    cy.wrap($item).click();
+  });
+});
+
+When('user clicks the {string} action of the context menu', function (actionKey: string) {
+  cy.get(`[data-test="graph-node-context-menu"] [data-test="${actionKey}_action"]`).then($item => {
+    cy.wrap($item).click();
+  });
+});
+
+Then('user should see the confirmation dialog to delete all traffic routing', function () {
+  cy.get('[data-test=delete-traffic-routing-modal]').should('exist');
```

Comment:
Hmm, it depends on what you refer, probably.
In the DOM, the modal isn't there when it's closed. So, this should fail if for some reason the modal is not launched.

The react component is another thing. I didn't looked at the tooling, but may be you are right that it could be present, but invisible.

So, I think that since this is checking the DOM, `should('be.visible')` may probably fail waiting for the control to appear in the DOM but invisible??? I'm not sure about it's behavior for these cases.

Reply:
Given the title, "user should see...", `be.visible` would seem appropriate.  But being a modal I think it's ok as-is, because  existence and visibility should be equivalent.